File: exercises/small_problems/easy1/how_many.rb
vehicles = [
  'car', 'CAR', 'truck', 'car', 'SUV', 'truck',
  'motorcycle', 'motorcycle', 'car', 'truck'
]

def count_occurrences(vehicle_list)
  vehicle_hash = {}

  vehicle_list.each do |vehicle|
    vehicle.downcase!
    if !vehicle_hash.key?(vehicle)
      vehicle_hash[vehicle] = 1
    else
      vehicle_hash[vehicle] += 1
    end
  end
  vehicle_hash.each { |k, v| puts k + " => " + v.to_s}
end


count_occurrences(vehicles)

File: exercises/jsSmallProblems/easy4/howMany.js
const vehicles = ['car', 'car', 'truck', 'car', 'SUV', 'truck',
  'motorcycle', 'motorcycle', 'car', 'truck'];

const countOccurrences = (arr) => {
  const result = {};

  arr.forEach((vehicle) => {
    result[vehicle] = result[vehicle] + 1 || 1;
  });

  Object.keys(result)
    .forEach((vehicle) => console.log(`${vehicle} => ${result[vehicle]}`));

  return result;
};

countOccurrences(vehicles);

/* console output
car => 4
truck => 3
SUV => 1
motorcycle => 2
*/
